presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/aggregation/GroupedGeometryOperatorState.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.geospatial.aggregation;
+
+import com.esri.core.geometry.GeometryCursor;
+import com.esri.core.geometry.ListeningGeometryCursor;
+import com.esri.core.geometry.SpatialReference;
+import com.esri.core.geometry.ogc.OGCGeometry;
+import com.facebook.presto.array.ObjectBigArray;
+import com.facebook.presto.spi.function.GroupedAccumulatorState;
+
+import static com.facebook.presto.plugin.geospatial.aggregation.GeometryCursorUtils.processGeometry;
+
+public class GroupedGeometryOperatorState
+        implements GeometryOperatorState.GeometryUnionOperatorState, GeometryOperatorState.GeometryConvexHullOperatorState, GroupedAccumulatorState
```

Comment:
Hmm... this looks strange. This class shouldn't depend on all possible aggregate functions.

Reply:
It's an artifact of using the annotations based aggregation framework (if I want to use the same state class for both types, then they must be instances of both interfaces).  I think I can remove this if I manually override `SqlAggregationFunction`, which I think will simplify other related things as well.